src/ophys_etl/modules/neuropil_correction/utils.py

```diff
@@ -0,0 +1,469 @@
+import logging
+from pathlib import Path
+from typing import List, Tuple, Union
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.sparse as sparse
+from scipy.linalg import solve_banded
+
+matplotlib.use("agg")
+
+
+def debug_plot(
+    file_name: Union[str, Path],
+    roi_trace: np.ndarray,
+    neuropil_trace: np.ndarray,
+    corrected_trace: np.ndarray,
+    r: float,
+    r_vals: np.ndarray = None,
+    err_vals: np.ndarray = None,
+) -> None:
+    """Create debug plot for neuropil correction
+
+    Parameters
+    -----------
+    file_name: Union[str, Path]
+        path to output plot image
+    roi_trace: np.ndarray
+        roi trace for a single roi
+    neuropil_trace: np.ndarray
+        neuropil trace for a single roi
+    corrected_trace: np.ndarray
+        corrected trace for a single roi
+    r: float
+        r contamination value
+    r_vals: np.ndarray
+        range of r values fitted
+    err_vals: np.ndarray
+        error values associated with range of r values
+    """
+    fig = plt.figure(figsize=(15, 10))
+
+    ax = fig.add_subplot(211)
+    ax.plot(roi_trace, "r", label="raw")
+    ax.plot(corrected_trace, "b", label="fc")
+    ax.plot(neuropil_trace, "g", label="neuropil")
+    ax.set_xlim(0, roi_trace.size)
+    ax.set_title(
+        "raw(%.02f, %.02f) fc(%.02f, %.02f) r(%f)"
+        % (
+            roi_trace.min(),
+            roi_trace.max(),
+            corrected_trace.min(),
+            corrected_trace.max(),
+            r,
+        )
+    )
+    ax.legend()
+
+    if r_vals is not None:
+        ax = fig.add_subplot(212)
+        ax.plot(r_vals, err_vals, "o")
+
+    plt.savefig(file_name)
+    plt.close()
+
+
+def get_diagonals_from_sparse(mat: sparse) -> dict:
```

Comment:
Could you please mark as private any functions not exposed outside of this module?